### PR TITLE
 spline: fix wrong start param after interpolation

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

**Comment by doudou** [comment]
@renan028 I believe this was a bug you had at some point ...

**Comment by renan028** [comment]
In ruby, I fixed a conversion https://github.com/ThirteenLtda/bundles-squidbot/commit/4772b78bb6de88b472afaec3f5a658fb7be4b23c
From 3 spline 1 to 1 Spline3, adding the parameters (times). I also had a problem appending two splines with different orders. But it is a different issue

File: src/Spline.cpp [hunk 1]
```diff
@@ -259,16 +259,21 @@ bool SplineBase::isNURBS() const
...
+                             vector< double >& parametersOut,
+                             const vector< double >& parametersIn,
                              const vector< SplineBase::CoordinateType >& coord_types)
 {
     clear();
     start_param = 0.0;
     has_curvature_max = false;
 
     int const point_count = points.size() / dimension;
+    if (!parametersIn.empty() && parametersIn.size() != point_count) {
```
[the remark this answers is not in the document]
Comment (doudou) [reply src/Spline.cpp]: Done.

File: src/Spline.cpp [hunk 2]
```diff
@@ -259,16 +259,21 @@ bool SplineBase::isNURBS() const
...
                              const vector< SplineBase::CoordinateType >& coord_types)
 {
     clear();
     start_param = 0.0;
     has_curvature_max = false;
 
-    int const point_count = points.size() / dimension;
+    unsigned int const point_count = points.size() / dimension;
+    if (!parametersIn.empty() && parametersIn.size() != point_count) {
+        throw std::invalid_argument(
```
Comment (renan028) [inline src/Spline.cpp]: Hi @doudou ,
I am making a spline given position and velocity:
{base::geometry::SplineBase::CoordinateType::ORDINARY_POINT,  base::geometry::SplineBase::CoordinateType::DERIVATIVE_TO_PRIOR}
I have point_count = 2 * n, where n is the number of parameters. And I am getting this error.

**Pull request merged**